packages/api-gateway/src/services/authorizationService.ts

```diff
@@ -93,49 +94,62 @@ async function isAllowedToGetClient(
     return true;
   }
 
-  const settledPromises = await Promise.allSettled(
-    client.purposes.map((purpose) =>
-      purposeProcessClient.getPurpose({
-        headers,
-        params: {
-          id: purpose,
-        },
-      })
-    )
-  );
+  return match(client)
+    .with(
+      { visibility: authorizationApi.ClientVisibility.Enum.FULL },
+      async (client) => {
+        const settledPromises = await Promise.allSettled(
+          client.purposes.map((purpose) =>
+            purposeProcessClient.getPurpose({
+              headers,
+              params: {
+                id: purpose,
+              },
+            })
+          )
+        );
 
-  settledPromises.forEach((p) => {
-    if (p.status !== "rejected") {
-      return;
-    }
-    /**
-     * There could be purposes which the requester is not allowed to see in the client.
-     * We ignore those purposes and continue the checks only with the ones that are allowed.
-     */
-    const error = p.reason;
-    if (isAxiosError(error) && error.response?.status === 403) {
-      return;
-    }
-    throw error;
-  });
+        settledPromises.forEach((p) => {
+          if (p.status !== "rejected") {
+            return;
+          }
+          /**
+           * There could be purposes which the requester is not allowed to see in the client.
+           * We ignore those purposes and continue the checks only with the ones that are allowed.
+           */
+          const error = p.reason;
+          if (isAxiosError(error) && error.response?.status === 403) {
+            return;
+          }
+          throw error;
+        });
 
-  const purposes = settledPromises
-    .filter(
-      (r): r is PromiseFulfilledResult<purposeApi.Purpose> =>
-        r.status === "fulfilled"
-    )
-    .map((r) => r.value);
+        const purposes = settledPromises
+          .filter(
+            (r): r is PromiseFulfilledResult<purposeApi.Purpose> =>
+              r.status === "fulfilled"
+          )
+          .map((r) => r.value);
 
-  const eservices = await Promise.all(
-    purposes.map((purpose) =>
-      catalogProcessClient.getEServiceById({
-        headers,
-        params: {
-          eServiceId: purpose.eserviceId,
-        },
-      })
-    )
-  );
+        const eservices = await Promise.all(
+          purposes.map((purpose) =>
+            catalogProcessClient.getEServiceById({
+              headers,
+              params: {
+                eServiceId: purpose.eserviceId,
+              },
+            })
+          )
+        );
 
-  return eservices.some((eservice) => eservice.producerId === requesterId);
+        return eservices.some(
+          (eservice) => eservice.producerId === requesterId
+        );
+      }
+    )
+    .with(
+      { visibility: authorizationApi.ClientVisibility.Enum.COMPACT },
+      () => false
```

Comment:
in this branch the requester could be the producer, which should be allowed to retrieve the client

Reply:
Broader question: the API GW V1 already returns a Client with only details present in the compact client, no matter the caller... could we just remove this visibility check and always return the Client? 
https://github.com/pagopa/interop-be-monorepo/blob/0479cadd807a4598be70ec9714f40054de89e2a4/packages/api-clients/open-api/apiGatewayApi.yml#L2511

Reply:
Settled in a call: let's remove this check, the API GW returns only the details we decided to make public to all tenants anyway (the client ID and the consumer ID). See [thread](https://pagopaspa.slack.com/archives/C08J74Y0H96/p1751533541560289).